src/main/java/org/openrewrite/java/spring/doc/MigrateDocketBeanToGroupedOpenApiBean.java

```diff
@@ -0,0 +1,319 @@
+                                        methodTemplateBuilder.append(".pathsToMatch(").append(paths).append(")\n");
+                                    }
+                                    if (docketDefinition.apis != null) {
+                                        String apis = docketDefinition.apis instanceof J.Literal ? "\"" + docketDefinition.apis + "\"" : docketDefinition.apis.toString();
+                                        methodTemplateBuilder.append(".packagesToScan(").append(apis).append(")\n");
+                                    }
+                                }
+                                methodTemplateBuilder.append(".build();\n")
+                                        .append("}");
+
+                                maybeAddImport("org.springdoc.core.models.GroupedOpenApi", false);
+                                maybeAddImport("org.springframework.context.annotation.Bean", false);
+                                return JavaTemplate.builder(methodTemplateBuilder.toString())
+                                        .contextSensitive()
+                                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "spring-context", "springdoc-openapi-starter-common"))
+                                        .imports("org.springdoc.core.models.GroupedOpenApi", "org.springframework.context.annotation.Bean").build()
+                                        .apply(getCursor(), method.getCoordinates().replace());
+                            }
+                            return md;
+                        }
+                    }.visitNonNull(tree, ctx);
+                } else if (isApplicationProperties(tree) && canConfigureInProperties(acc, docketDefinition)) {
+                    Tree result = addSpringProperty(ctx, tree, "springdoc.api-docs.path", "/v3/api-docs");
+                    result = addSpringProperty(ctx, result, "springdoc.swagger-ui.path", "/swagger-ui.html");
+                    if (docketDefinition.groupName == null) {
+                        result = formatGroupProperties(ctx, result, "springdoc", docketDefinition);
+                    } else {
+                        result = addSpringProperty(ctx, result, "springdoc.group-configs[0]" + ".group", docketDefinition.groupName.toString());
+                        result = formatGroupProperties(ctx, result, "springdoc.group-configs[0]", docketDefinition);
+                    }
+                    return result;
+                }
+                return super.visit(tree, ctx, parent);
+            }
+        });
+    }
+
+    private static boolean isApplicationProperties(@Nullable Tree tree) {
+        return (tree instanceof Properties.File && "application.properties".equals(((Properties.File) tree).getSourcePath().getFileName().toString())) ||
+                (tree instanceof Yaml.Documents && ((Yaml.Documents) tree).getSourcePath().getFileName().toString().matches("application\\.ya*ml"));
```

Comment:
Would we want to glance at or even edit any [profile specific properties files](https://docs.spring.io/spring-boot/reference/features/external-config.html#features.external-config.files.profile-specific)?